core/profiles/src/main/java/software/amazon/awssdk/profiles/internal/ProfileFileReader.java

```diff
@@ -44,6 +44,8 @@
 
     private static final Pattern VALID_IDENTIFIER = Pattern.compile("^[A-Za-z0-9_\\-]*$");
 
+    private static final Pattern VALID_PROFILE_NAME = Pattern.compile("^[A-Za-z0-9_\\-/.%@]*$");
```

Comment:
It's probably easier to just update the identifier pattern for this. Thoughts?

Reply:
Do you want these characters to be supported in the property names? I didn't think we should support them in property names. I'll do what you want, though.

Reply:
I don't think we'll ever create properties with those characters in the name. I don't think it hurts, though, and it keeps things simpler.

Reply:
OK. I've updated in d778c39974 .